curve_parameters/new_curve_script.py

```diff
@@ -14,25 +17,28 @@ def to_hex(val: int, length):
     n = 8
     chunks = [x[i:i+n] for i in range(0, len(x), n)][::-1]
     s = ""
-    for c in chunks:
+    for c in chunks[:length // n]:
```

Comment:
Why is this necessary?

Reply:
It's for a weird corner case. Most of the time, `length // n` is the length of `chunks`, but sometimes the number `val` passed into `to_hex` function can be larger than expected. For example, `4 * modulus` might be 257 bits long for 255-bit moduli. In which case an extra limb will be added leading to a compile error. Of course, trimming `chunks` like I'm doing is also unsafe - if `4 * modulus` in the above example is used at any point, this will lead to a silent failure during runtime. Currently, the only field with this issue is the scalar field of BLS12-381 curve which doesn't use `4 * modulus` anywhere. But in principle this is problematic and we should probably rewrite `field.cu` in a way that allows removing such overflowing values from the params file with compile time errors when trying to use them